src/psyclone/tests/domain/nemo/transformations/nemo_allarrayrange2loop_trans_test.py

```diff
@@ -157,6 +157,40 @@ def test_apply_with_structures(fortran_reader, fortran_writer):
     assert ("ptab(jf)%pt2d(jpi,idx_1,idx) = "
             "ptab(jf)%pt2d(jpim1,idx_1,idx)") in result
 
+def test_apply_option_verbose(fortran_reader, capsys):
+    '''Check that the transformation with the verbose option provides more
+    information about why the last attempt to convert the array range into
+    a loop was not successful.
+
+    '''
+    trans = NemoAllArrayRange2LoopTrans()
+
+    # The outer dimension is already set
+    psyir = fortran_reader.psyir_from_source('''
+    subroutine test
+        use my_variables
+        integer, parameter :: constant = 3
+        integer :: jk, a, b
+        integer, dimension(3) :: array
+        a = b
+        array(:) = my_func()
+    end subroutine test
+    ''')
+
+    # Basic cases like assingment 1 are skipped
+    assignment = psyir.walk(Assignment)[0]
+    trans.apply(assignment, options={'verbose': True})
+    out, _ = capsys.readouterr()
+    assert out == ""
+
+    # Other cases liek assingment 2 are printed to stdout
```

Comment:
"like". Do we get a CodeBlock because we don't know how `my_func` is brought into scope? Please add this to the comment.